src/services/dashboards/dashboard-customize/modules/DashboardAddDefaultWidgetRightArea.vue

```diff
@@ -108,20 +198,118 @@ export default defineComponent<Props>({
             name(value: string) { return value.trim().length ? '' : i18n.t('Please enter name.'); },
         });
 
+        const {
+            targetRef, targetElement, contextMenuStyle,
+        } = useContextMenuFixedStyle({
+            useFixedMenuStyle: true,
+            visibleMenu: toRef(state, 'addOptionsMenuVisible'),
+        });
+        const contextMenuFixedStyleState = reactive({
+            targetRef, targetElement, contextMenuStyle,
+        });
+
+        /* Util */
+        const isSelected = (selectedItem) => {
+            if (Array.isArray(selectedItem)) return !!selectedItem.length;
+            return selectedItem && !isEmpty(selectedItem);
+        };
+        const getRefinedJsonSchema = (originalJsonSchema: JsonSchema, selectedOptions, inheritItemMap) => {
```

Comment:
This function rendering refined jsonSchema from widget config.
For example,
```typescript
// widget-config.ts
{
  ...
  schema: {
    type: 'object',
    properties: {
      provider: {
        title: 'Provider',
        type: 'array',
        enum: ?? // we don't know here in widget-config.ts. because it's store data
        menuItems: ?? //
      }
    }
  }
}

// after refined with function
{
  ...
  schema: {
    type: 'object',
    properties: {
      provider: {
        title: 'Provider',
        type: 'array',
        enum: ['aws', 'google', 'azure', ...]
        menuItems: [{ name: 'aws', label: 'AWS' }, ...]
      }
    }
  }
}

```